src/compiler/commandLineParser.ts

```diff
@@ -547,10 +547,14 @@ namespace ts {
         }
     }
 
-    export function convertCompilerOptionsFromJson(jsonOptions: any, basePath: string): { options: CompilerOptions, errors: Diagnostic[] } {
+    export function convertCompilerOptionsFromJson(jsonOptions: any, basePath: string, configFileName?: string): { options: CompilerOptions, errors: Diagnostic[] } {
         const options: CompilerOptions = {};
         const errors: Diagnostic[] = [];
 
+        if (configFileName && getBaseFileName(configFileName) === "jsconfig.json") {
+            options.allowJs = true;

```

Comment:
Default "module" to "commonjs" also for jsconfig.json files, as by default we infer CommonJS from `require` calls or `exports` assignments, and also this will allow usage of ES6 syntax without getting errors about setting the module type (which is mostly needed for emit, which usually doesn't apply in the .js case).